spring-ai-core/src/main/java/org/springframework/ai/document/ContentFormatter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.document;
+
+import java.util.function.Function;
+
+/**
+ * Converts the Document text and metadata into a AI, prompt-friendly text representation.
+ *
+ * @author Christian Tzolov
+ */
+public interface ContentFormatter extends Function<Document, String> {
+
+	enum MetadataMode {
+
+		ALL, EMBED, LLM, NONE;
```

Comment:
I think we should change LLM to something like `INFERENCE` of `MODEL` as I've removed use of `LLM` generally speaking in naming things.

Reply:
I like INFERENCE